source/request-as-event-emitter.ts

```diff
@@ -72,20 +82,21 @@ export default (options, input?: TransformStream) => {
 
 		if (agents) {
 			const protocolName = options.protocol === 'https:' ? 'https' : 'http';
-			options.agent = agents[protocolName] || options.agent;
+			// TS complains due to index signature missing
```

Comment:
Is this a problem with our types or the Node.js types?

Reply:
It has been solved! 🎉